lib/active_model/serializer.rb

```diff
@@ -338,7 +338,7 @@ def to_json(*args)
     # object including the root.
     def as_json(options={})
       options ||= {}
-      if root = options.fetch(:root, @options.fetch(:root, root_name))
+      if root = options.to_hash.fetch(:root, @options.fetch(:root, root_name))
```

Comment:
@bf4 I fixed some of your gem's tests, so this is now in a better state than the current 0-8-stable branch.  It's not passing on some older versions of ruby that you guys are testing on travis because:

> mime-types requires Ruby version >= 1.9.2

[as seen here](https://travis-ci.org/rails-api/active_model_serializers/builds/89587378).